x-pack/test/security_solution_endpoint/services/endpoint_policy.ts

```diff
@@ -48,14 +48,61 @@ export interface PolicyTestResourceInfo {
 export function EndpointPolicyTestResourcesProvider({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
   const log = getService('log');
+  const retry = getService('retry');
 
-  const logSupertestApiErrorAndThrow = (message: string, error: any) => {
+  const logSupertestApiErrorAndThrow = (message: string, error: any): never => {
     const responseBody = error?.response?.body;
     const responseText = error?.response?.text;
+    log.error(`Error occurred at ${Date.now()} | ${new Date().toISOString()}`);
     log.error(JSON.stringify(responseBody || responseText, null, 2));
     log.error(error);
     throw new Error(message);
   };
+  const retrieveEndpointPackageInfo = (() => {
+    // Retrieve information about the Endpoint security package
+    // EPM does not currently have an API to get the "lastest" information for a page given its name,
+    // so we'll retrieve a list of packages for a category of Security, and will then find the
+    // endpoint package info. in the list. The request is kicked off here, but handled below after
+    // agent config creation so that they can be executed concurrently
+    let apiRequest: Promise<GetPackagesResponse['response'][0] | undefined>;
```

Comment:
api Request or Response?

Reply:
`apiRequest` is the promise that makes the request and returns the response. I can see the confusion 😃 .
Maybe I should rename it `packageInfoResponse`?